automation/workflow/hook.go

```diff
@@ -0,0 +1,24 @@
+package workflow
+
+import "github.com/qri-io/qri/event"
+
+// "encoding/json"
+
+// HookType is the type of hook
+type HookType string
+
+// A Hook determines under what circumstances its `hook.Event()` should be
+// emitted, and what the event payload should be.
+type Hook interface {
+	// json.Marshaler
+	// json.Unmarshaler
+	ID() ID
+	IsEnabled() bool
+	Enable() error
+	Disable() error
+	Type() HookType
+	ToMap() map[string]interface{}
```

Comment:
I'd love a comment here:
```go
// ToMap serializes all hook state
ToMap() map[string]interface{}
```

I'm confused about the need for this method, given the presence of the `Payload`  method and the planned requirement for json serialization. If I can get a hook to go to-and-from JSON, and I'm not using `ToMap` as the payload of an event, what do I need `ToMap` for?

Reply:
Not sure yet, lol. This was holdover from our previous iteration, and I held onto it while planning in case I was missing something. But I agree, with the plans for serialization what's the point?